huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_mysql_binlog_test.go

```diff
@@ -0,0 +1,158 @@
+	region := acceptance.HW_REGION_NAME
+
+	var (
+		mysqlBinlogHttpUrl = "v3/{project_id}/instances/{instance_id}/binlog/clear-policy"
+		mysqlBinlogProduct = "rds"
+	)
+	mysqlBinlogClient, err := cfg.NewServiceClient(mysqlBinlogProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating RDS client: %s", err)
+	}
+	instanceID := state.Primary.Attributes["instance_id"]
+	mysqlBinlogPath := mysqlBinlogClient.Endpoint + mysqlBinlogHttpUrl
+	mysqlBinlogPath = strings.ReplaceAll(mysqlBinlogPath, "{project_id}", mysqlBinlogClient.ProjectID)
+	mysqlBinlogPath = strings.ReplaceAll(mysqlBinlogPath, "{instance_id}", instanceID)
+	mysqlBinlogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+	mysqlBinlogResp, err := mysqlBinlogClient.Request("GET", mysqlBinlogPath, &mysqlBinlogOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS Mysql binlog")
+	}
+	mysqlBinlogRespBody, err := utils.FlattenResponse(mysqlBinlogResp)
+	if err != nil {
+		return nil, err
+	}
+	binlog := utils.PathSearch("binlog_retention_hours", mysqlBinlogRespBody, nil)
+	if binlog == nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return mysqlBinlogRespBody, nil
+}
+
+func TestAccMysqlBinlog_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_rds_mysql_binlog.test"
+	dbPwd := fmt.Sprintf("%s%s%d", acctest.RandString(5),
+		acctest.RandStringFromCharSet(2, "!#%^*"), acctest.RandIntRange(10, 99))
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getMysqlBinlogResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testMysqlBinlog_basic(name, dbPwd),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_rds_instance.test", "id"),
+					resource.TestCheckResourceAttr(rName, "binlog_retention_hours", "6"),
+				),
+			},
+			{
+				Config: testMysqlBinlog_basic_update(name, dbPwd),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_rds_instance.test", "id"),
+					resource.TestCheckResourceAttr(rName, "binlog_retention_hours", "8"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccRdsInstance_mysql(name, pwd string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_rds_flavors" "test" {
+db_type       = "MySQL"
+db_version    = "8.0"
+instance_mode = "single"
+group_type    = "dedicated"
+}
```

Comment:
```suggestion
data "huaweicloud_rds_flavors" "test" {
  db_type       = "MySQL"
  db_version    = "8.0"
  instance_mode = "single"
  group_type    = "dedicated"
}
```
and fix other places

Reply:
done